### ISSUE TITLE
Add Manifest Uses Permission Automatically

### ISSUE THREAD

**Issue by nespapu** [phearme/cordova-ContentProviderPlugin#1]
Hi!

First of all, thank you very much for this plugin.

We are using it to build an application. We will use phonegap remote build services to compile the app. Due to this we have face a problem: the uses-permission denoting the package name of the application that expose the content provider cannot be added manually.

I have been googling for a few hours trying to figure out how this plugin can be modify to do that, but I have had no luck. Other possible solution is to add the permission through the config.xml file but all is very confusing and I neither find the way.

Maybe you know how to do that?

Thanks again for your plugin.

Regards,
Néstor

**Comment by phearme** [comment]
Hi Néstor.
You can add permissions through the config.xml file. As an exemple, I will put a comment inside the config.xml used in this plugin, and push it back. Will let you know once it's done.

**Comment by phearme** [comment]
Just added the example. Let me know if that works for you ;-)

**Comment by nespapu** [comment]
Thank you very much for your quick response!

Unfortunatly, I could not get it work following your instruction. My config.xml remains this way:

**Comment by phearme** [comment]
Ok. You could also fork the sources of the current plugin, edit the plugin.xml file to add the permissions you need, publish the plugin and use it in your project. You would need to publish your plugin on a public repo (such as github) if you wish to build your app through Phonegap Build (I guess).

**Comment by nespapu** [comment]
Hello again!

Finally I got it working! :) 

The solution you pointed was right. This snippet works!

```
 <gap:config-file platform="android" parent="/manifest" mode="add">
        <uses-permission android:name="com.some.app.permission.READ" />
 </gap:config-file>
```
**The key was that this tag only works when you use the phonegap build services, If you build it locally it will never take effect.** I lost a lot of time until I found a post pointing that. Anyway, all is good now.

Thank you again for your work.

Cheers!
